src/HypothesisClient/Client/Search.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace eLife\HypothesisClient\Client;
+
+use eLife\HypothesisClient\ApiClient\SearchClient;
+use eLife\HypothesisClient\Model\Annotation;
+use eLife\HypothesisClient\Result\Result;
+use GuzzleHttp\Promise\PromiseInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class Search
+{
+    private $total = [];
+    private $serializer;
+    private $searchClient;
+
+    public function __construct(SearchClient $searchClient, DenormalizerInterface $serializer)
+    {
+        $this->searchClient = $searchClient;
+        $this->serializer = $serializer;
+    }
+
+    public function query(
+        string $username = null,
+        string $accessToken = null,
+        int $offset = 0,
+        int $limit = 20,
+        bool $descendingOrder = true,
+        string $sort = 'updated'
+    ) : PromiseInterface {
+        return $this->searchClient
+            ->query(
+                [],
+                $username,
+                $accessToken,
+                $offset,
+                $limit,
+                $descendingOrder,
+                $sort
+            )
+            ->then(function (Result $result) use ($username, $accessToken) {
+                $this->total[$this->getTotalKey($username, $accessToken)] = $result['total'];
+
+                return $result;
+            })
+            ->then(function (Result $result) {
+                return array_map(function (array $annotation) {
+                    return $this->serializer->denormalize($annotation, Annotation::class);
+                }, $result['rows']);
+            });
+    }
+
+    public function count(
```

Comment:
the two methods can be called in any order, which reflect statelessness. `$this->total` is however a cache that is never invalidated, is it strictly necessary or is it a performance optimization?

Reply:
The only values that should impact the total are `$username` and `$accessToken`. We do not hold on to access tokens. It is not necessary, just a performance optimisation.

Reply:
Also time passing impacts the value e.g. if a process holds onto an instance of this class it doesn't see the count growing
So like all layers of caching I'd drop it until its need is measured (perhaps the API is so slow that we'll need another layer of caching anyway, or it's so fast that is unnecessary)